### PR TITLE
Avoid sending 304s in Symfony kernel.debug mode

### PR DESCRIPTION
**Pull request by mpdude**
This may be confusing for developers, since changes to e. g. Twig templates may not become visible, or requests even completely bypass controllers, changes to DIC configuration or similar.

Almost the same code as we have here:

https://github.com/webfactory/WebfactoryWfdMetaBundle/blob/7592793915d7d9b096a6c49bbdc06acae1bd7e90/Caching/EventListener.php#L60-L70

### CODE REVIEW

File: NotModified/EventListener.php [hunk 1]
```diff
@@ -108,7 +120,7 @@ private function findAnnotation(callable $controllerCallable)
             return null;
         }
 
-        list($class, $methodName) = $controllerCallable;
+        [$class, $methodName] = $controllerCallable;
```
Comment (MalteWunsch) [inline NotModified/EventListener.php:123]: This short syntax requires PHP 7.1 - should we require it in composer.json oder revert the change? I'm fine either way.

Comment (mpdude) [reply NotModified/EventListener.php:123]: I don’t think we need to support < PHP 7.1, so bumping the requirement would be fine. Thanks for catching this!
